Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
+            (req.required? || build.with?(req))
+        end ||
+          deps.any? do |dep|
+            dep.name == "python" && (dep.required? || build.with?(dep))
+          end
+      end
```

Comment:
The problem with this method is that absence of any Python 2 requirement or hard-coded dependency doesn't mean the formula doesn't support or want the system Python, thus `python2`. Even checking the stand-alone `--with-python` or `--without-python` options isn't sufficient, unless I'm misreading [`Language::Python.each_python`](https://github.com/Homebrew/brew/blob/38209aadbfe4fd0c6772467c4bc5c63325d53f6c/Library/Homebrew/language/python.rb#L16-L29). That means it should be legal that a formula specifies

```ruby
depends_on :python if MacOS.version <= :snow_leopard
depends_on :python3
```

with the intention of always providing both Python 2 and Python 3 support.